### DISCUSSION TITLE
PR Template structure

### DISCUSSION THREAD

**Discussion by danielRep** [bao-project/bao-ci#64] [General]
The PR templates where created to format and detail a PR description. For now, we have main three "sections" to be addressed:

- PR Description
- Type of Change: to signalize the type of change the PR introduces 
- Checklist: a list that forces contributors to perform a set of tasks before sending the PR (run CI checkers, build without errors/warnings, etc).

@josecm raised the concern in https://github.com/bao-project/bao-ci/issues/35#issuecomment-1474876534 that such full description can be overkill. I open this discussion so we can discuss how can we improve the template.

**Comment by danielRep** [comment]
Now that we have been using this for sometime, I think the type of change can be redundant. Most of the times we clearly identify the "type of change" on the PR subject given that github gathers the branch name to the subject. Regarding the "PR Description", I think it's consensual and the checklist we can reorganize in another way.

**Reply by DavidMCerdeira** [reply]
> Most of the times we clearly identify the "type of change" on the PR subject given that github gathers the branch name to the subject. 

Agree

>Regarding the "PR Description", I think it's consensual and the checklist we can reorganize in another way.

Yeah the "tasks" are helpful to remind the PR author to perform some steps, but I agree that it is not necessary to be part of the description.

Do you have any specific ideas on how to reorganize the template description?

**Reply by danielRep** [reply]
I don't have a clear idea on how to organize. Any you guys have good examples?

**Comment by danielRep** [comment]
ping @bao-project/maintainers

**Comment by danielRep** [comment]
New PR template proposal @bao-project/maintainers :
- remove the type of change. Seems redundant most of the times, since the specific change is typically specified in the PR subject
- add a small intro text, pointing to quick tips before submitting the PR. This in essence substitutes the checklist
- remove the checklist, since it seems no useful.
- the tag homu-ignore makes sure that even if the contributor forgets to remove that extra information, it doesn't not show on the PR description when submitted. 

**PROPOSAL:** 
```
<!-- homu-ignore:start -->
<!--
```
Thanks for submitting a PR to the Bao Project :tada: ! Before submit the PR, please be sure you have read the next tips to ensure your PR is reviewed and merged faster.
- If you are submitting your first PR, you should definitely read our [contributing guidelines](https://github.com/bao-project/bao-docs/blob/main/source/development/contributing.rst): 
- If you are submitting a change on documentation, be certain it follows [documentation guidelines](https://github.com/bao-project/bao-docs/blob/main/source/development/doc_guidelines.rst)
- If you are submitting code changes, be certain it follows [coding guidelines](https://github.com/bao-project/bao-docs/blob/main/source/development/code_documentation.rst) and [style](https://github.com/bao-project/bao-docs/blob/main/source/development/coding_style.rst), and [MISRA](https://github.com/bao-project/bao-docs/blob/main/source/development/misra.rst) if applicable 
-  
```
-->
<!-- homu-ignore:end -->
```
**PR Description**
```
<!-- homu-ignore:start -->
<!--
```
Please include a summary of the change that tries to answer the following general questions:
- What does this change introduces? Why is it required?
- Which main blocks are affected?
If your changes are included in the following categories, please try also to answer to these specific questions if they enhance your summary:
- The PR changes introduces a new feature...
  - What new feature is being introduced?
  - Is the feature described in any of the requirements?
- The PR changes solves a bug...
  - What bug does it solve? What was the previous behavior/warning/error?
  - Is the bug described in any issue ticket number?
- The PR changes refactors a code/doc block...
  - Why the code/doc block needed to be refactored?
  - Is the improvement described in any issue ticket number?
- The PR changes the build system...
  - What building block of the build system was changed?
  - Has any external dependency introduced?
- The PR changes improves the overall performance of the system...
  - Did you quantify the improvement on performance? Please share the numbers.
- The PR changes or adds new tests...
  - What new tests were introduced or corrected?
```
-->
<!-- homu-ignore:end -->
```

**Testing the PR** 
```
<!-- homu-ignore:start -->
<!--
```
Explain on your changes should be tested and guide the reviewers on how to do it. Moreover, explain which tests have been run, or if any tests should be added to the bao test framework. 
```
-->
<!-- homu-ignore:end -->
```

**Additional information**
```
<!-- homu-ignore:start -->
<!--
```
Other information you want to mention in this PR, such as a TODO list. If you are opening the base PR of a chain of PRs, please add here a list of the next features/changes to be added on the following chained PRs. Try to update the list as each PR is submitted. 
```
-->
<!-- homu-ignore:end -->
```

**Reply by danielRep** [reply]
I could also add a draft table for wip branches of large PRs (e.g. #68).

**Reply by DavidMCerdeira** [reply]
> Explain on your changes should be tested and guide the reviewers on how to do it. Moreover, explain which tests have been run, or if any tests should be added to the bao test framework.

We should add at least one test for each new added feature / modification if not covered already by previous tests.
For the time being we should disregard the test framework requirement, and revisit it when it is up and running.

**Reply by DavidMCerdeira** [reply]
This is what I envisioned the template to be, good job!
